src/external.py

```diff
@@ -0,0 +1,61 @@
+# Copyright (c) 2018-2020, Michael P. Howard
+# Copyright (c) 2021-2022, Auburn University
+# This file is part of the azplugins project, released under the Modified BSD License.
+
+
+import hoomd
+
+from hoomd.azplugins import _azplugins
+from hoomd.data.parameterdicts import TypeParameterDict
+from hoomd.data.typeparam import TypeParameter
+from hoomd.data.parameterdicts import ParameterDict
+from hoomd.data.typeconverter import OnlyTypes, variant_preprocessing
+from hoomd.md import force
+
+class HarmonicBarrier(force.Force):
```

Comment:
You should make these separate classes that mirror the C++ side (`PlanarHarmonicBarrier` and `SphericalHarmonicBarrier`). They can both derive from a common `HarmonicBarrier` class though! You might follow a pattern like I did here:
https://github.com/mphowardlab/azplugins/blob/fb0ac2fe6cb894e8a437e934f8197897f3d6ad49/src/compute.py#L17